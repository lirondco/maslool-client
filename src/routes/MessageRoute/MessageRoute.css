.MessageRoute {
    margin: 2%;
    padding: 2%;
  }
  
  .MessageRoute h2 {
    margin-bottom: 1%;
  }
  
  hr {
    border: 1px solid;
  }
  