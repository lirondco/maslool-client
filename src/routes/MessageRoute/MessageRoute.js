import React, { Component } from "react";
import { Link } from "react-router-dom";
import MessageContent from "../../components/MessageContent/MessageContent";
import PendingApiService from "../../services/pending-api-service";
import "./MessageRoute.css";

export default class MessageRoute extends Component {
  state = {
    pending: null,
  };

  static defaultProps = {
    match: { params: {} },
  };

  componentDidMount = () => {
    this.setState({ error: 'Fetching message ...', pending: null });
    const { pendingId } = this.props.match.params;
    PendingApiService.getPending(pendingId)
      .then((pending) => this.setState({ pending, error: null }))
      .catch((error) => this.setState({ ...error }));
  };

  handleDeleteMessage = () => {
    this.setState({ error: null });
    const { pendingId } = this.props.match.params;
    PendingApiService.deletePending(pendingId)
      .then(() => this.props.history.goBack())
      .catch((error) => this.setState({ error }));
  };

  renderMessage = () => {
    const { pending } = this.state;
      return <MessageContent pending={pending} onDeleteClick={this.handleDeleteMessage} />;
  };

  render() {
      const { error, pending } = this.state
    return (
      <section className="MessageRoute">
        {(!pending) ? <h2>Loading message...</h2> : <h2>Message from {pending.user.username}</h2>}
        <hr />
        <Link to='/messages'>
          <h3>{"<"} Back to Messages</h3>
        </Link> 
        <div role="alert">{error && <p className="error">{error}</p>}</div>
        {this.renderMessage()}
      </section>
    );
  }
}
