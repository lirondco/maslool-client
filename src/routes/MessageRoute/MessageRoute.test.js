import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MessageRoute from "./MessageRoute";

describe(`MessageRoute component`, () => {
  it("renders the complete message", () => {
    const wrapper = shallow(<MessageRoute />);
    wrapper.setState({
      pending:
        {
          id: 9,
          message: "test",
          user: {
            id: 1,
            username: "spongebob",
            email: "sb.squarepants@bikinibottom.net",
            join_date: "2021-03-11T06:47:06.477Z",
            admin: false,
            banned: false,
          },
          error: null
        },
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
