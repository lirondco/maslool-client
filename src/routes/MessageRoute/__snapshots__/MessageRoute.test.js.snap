// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageRoute component renders the complete message 1`] = `
<section
  className="MessageRoute"
>
  <h2>
    Message from 
    spongebob
  </h2>
  <hr />
  <Link
    to="/messages"
  >
    <h3>
      &lt;
       Back to Messages
    </h3>
  </Link>
  <div
    role="alert"
  >
    <p
      className="error"
    >
      Fetching message ...
    </p>
  </div>
  <MessageContent
    onDeleteClick={[Function]}
    pending={
      Object {
        "error": null,
        "id": 9,
        "message": "test",
        "user": Object {
          "admin": false,
          "banned": false,
          "email": "sb.squarepants@bikinibottom.net",
          "id": 1,
          "join_date": "2021-03-11T06:47:06.477Z",
          "username": "spongebob",
        },
      }
    }
  />
</section>
`;
